### ISSUE TITLE
Passing locals dict to sympify

### ISSUE THREAD

**Issue by zaprice** [symengine/symengine.py#321]
When constructing an expression from a string in SymPy, it is often useful to pass around a `locals` dict defining necessary symbols. This functionality is present in `sympy.sympify` via the argument `locals`, and in `sympy.parsing.sympy_parser.parse_expr` via the argument `local_dict`.

Rather than declaring them in the local environment and constructing an expression with `eval`, we can pass around a dict of symbols resulting in cleaner code (e.g. no linter complaints about variables only used inside `eval`, re-use is more straightforward, etc).

It would be nice to have this in symengine.py as well!

**Comment by isuruf** [comment]
Yes, we'd like to have this. Let me know if you'd like to work on this and I can give pointers and help you with it.
